src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -231,7 +230,6 @@ private enum Dependency
                 constructedContainingTypeOfMethod,
                 formalParameterTypes,
                 formalParameterRefKinds,
-                argumentTypes,
```

Comment:
Now it looks like this parameter is not used in any meaningful way. Any reason to keep it?